xarray/core/common.py

```diff
@@ -347,6 +347,34 @@ def _ipython_key_completions_(self) -> list[str]:
         return list(items)
 
 
+class TreeAttrAccessMixin(AttrAccessMixin):
```

Comment:
I initially tried to use `AttrAccessMixin`. Just dropping it in as a replacement for `TreeAttrAccessMixin` doesn't work because of the checking done inside the `AttrAccessMixin.__init_subclass__` method. The check for if `__dict__` exists fails. `DataTree`, `TreeNode` and `NamedNode` do all define `__slots__`, but I think there are attributes being defined in places like `DataTree.__init__`, for example `DataTree.parent` and `DataTree.children` (versus the `DataTree._parent` and `DataTree._children` attributes defined in `DataTree.__slots__`).

I think we talked about this a little bit in one of our calls - things like `DataTree.parent` and `DataTree.children` seemed like tricky things to change over.